File: tests/python/cases/case_113_closure_chain_async_02.py
"""
Task case 113: closure_chain_async_02 false-positive/false-negative stress pattern.
"""

RETAINED_CACHE = {}
RETAINED_AUDIT = []
RETAINED_HANDLERS = []

def case_113_closure_chain_async_02(input_data):
    task_name = 'closure_chain_async_02'
    raw = input_data or 'sample'
    payload = {
        'task': task_name,
        'entity': 'stress',
        'stage': 'evaluation',
        'input': raw,
        'checksum': f'{task_name}:{len(raw)}',
    }
    def handler() -> str:
        return payload['input']
    # ESCAPE: closure captures payload and retained handler outlives function scope.
    RETAINED_HANDLERS.append(handler)
    return 'ok'

File: tests/go/cases/case_113_closure_chain_async_02.go
package escape_tests

var retainedCase113 = []map[string]string{}

func Case113ClosureChainAsync02(input string) string {
	raw := input
	if raw == "" {
		raw = "sample"
	}
	payload := map[string]string{
		"task": "closure_chain_async_02",
		"entity": "stress",
		"stage": "evaluation",
		"input": raw,
		"checksum": "closure_chain_async_02:" + raw,
	}
	handler := func() string { return payload["input"] }
	_ = handler
	// ESCAPE: closure payload persisted indirectly via retained metadata map.
	retainedCase113 = append(retainedCase113, map[string]string{"h": payload["input"]})
	return "ok"
}
